conans/tools.py

```diff
@@ -392,47 +392,111 @@ def get_solaris_version_name(self, version):
         env_sudo = os.environ.get("CONAN_SYSREQUIRES_SUDO", None)
         self._sudo = (env_sudo != "False" and env_sudo != "0")
-        self._os_info = OSInfo()
+        os_info = os_info or OSInfo()
+        self._is_up_to_date = False
+        self._tool = tool or self._create_tool(os_info)
+        self._tool._sudo_str = "sudo " if self._sudo else ""
+        self._tool._runner = runner or ConanRunner()
+
+    def _create_tool(self, os_info):
+        if os_info.with_apt:
+            return AptTool()
+        elif os_info.with_yum:
+            return YumTool()
+        elif os_info.is_macos:
+            return BrewTool()
+        else:
+            return NullTool()
 
     def update(self):
         """
             Get the system package tool update command
         """
-        sudo_str = "sudo " if self._sudo else ""
-        update_command = None
-        if self._os_info.with_apt:
-            update_command = "%sapt-get update" % sudo_str
-        elif self._os_info.with_yum:
-            update_command = "%syum check-update" % sudo_str
-        elif self._os_info.is_macos:
-            update_command = "brew update"
-
-        if update_command:
-            print("Running: %s" % update_command)
-            if self._runner(update_command, True) != 0:
-                raise ConanException("Command '%s' failed" % update_command)
+        self._is_up_to_date = True
+        self._tool.update()
 
-    def install(self, package_name):
+    def install(self, packages, update=True, force=False):
         '''
             Get the system package tool install command.
         '''
-        sudo_str = "sudo " if self._sudo else ""
-        install_command = None
-        if self._os_info.with_apt:
-            install_command = "%sapt-get install -y %s" % (sudo_str, package_name)
-        elif self._os_info.with_yum:
-            install_command = "%syum install -y %s" % (sudo_str, package_name)
-        elif self._os_info.is_macos:
-            install_command = "brew install %s" % package_name
-
-        if install_command:
-            print("Running: %s" % install_command)
-            if self._runner(install_command, True) != 0:
-                raise ConanException("Command '%s' failed" % install_command)
-        else:
-            print("Warn: Only available for linux with apt-get or yum or OSx with brew")
-            return None
+        packages = [packages] if isinstance(packages, str) else list(packages)
+        if not force and self._installed(packages):
+            return
+        if update and not self._is_up_to_date:
+            self.update()
+        self._install_any(packages)
+
+    def _installed(self, packages):
+        for pkg in packages:
+            if self._tool.installed(pkg):
+                print("Package already installed: %s" % pkg)
+                return True
+        return False
+
+    def _install_any(self, packages):
+        if len(packages) == 1:
```

Comment:
Yes they are already managed by the for loop, however the error reporting is a bit different for one package vs multiple packages with this approach.
`installer.install('some_package')` will raise a more detailed error message than `installer.install(['some_package', 'some_package_1.2'])`.
This could be made more clear with a comment